htdocs/core/lib/functions.lib.php

```diff
@@ -12822,56 +12857,96 @@ function fetchObjectByElement($element_id, $element_type, $element_ref = '', $us
-		}
 
-		dol_include_once('/'.$element_prop['classpath'].'/'.$element_prop['classfile'].'.class.php');
+	if (!is_array($element_prop) || (!empty($element_prop['module']) && !$ismodenabled)) {
+		$errMsgCode = 'ElementHasNoPropertyORModuleNotEnabled';
+		return false;
+	}
 
-		if (class_exists($element_prop['classname'])) {
-			$className = $element_prop['classname'];
-			$objecttmp = new $className($db);
-			'@phan-var-force CommonObject $objecttmp';
+	dol_include_once('/'.$element_prop['classpath'].'/'.$element_prop['classfile'].'.class.php');
+	if (!class_exists($element_prop['classname'])) {
+		$errMsgCode = 'ElementClassNotExist';
+		return false;
+	}
 
-			if ($element_id > 0 || !empty($element_ref)) {
-				$ret = $objecttmp->fetch($element_id, $element_ref);
-				if ($ret >= 0) {
-					if (empty($objecttmp->module)) {
-						$objecttmp->module = $element_prop['module'];
-					}
+	$className = $element_prop['classname'];
+	$objecttmp = new $className($db);
+	'@phan-var-force CommonObject $objecttmp';
 
-					if ($useCache > 0) {
-						if (!isset($globalCacheForGetObjectFromCache[$element_type])) {
-							$globalCacheForGetObjectFromCache[$element_type] = [];
-						}
+	if (property_exists($objecttmp, 'module') && empty($className->module)) {
+		$objecttmp->module = $element_prop['module'];
+	}
 
-						// Manage cache limit
-						if (! empty($globalCacheForGetObjectFromCache[$element_type]) && is_array($globalCacheForGetObjectFromCache[$element_type]) && count($globalCacheForGetObjectFromCache[$element_type]) >= $maxCacheByType) {
-							array_shift($globalCacheForGetObjectFromCache[$element_type]);
-						}
+	return $objecttmp;
+}
 
-						$globalCacheForGetObjectFromCache[$element_type][$element_id] = $objecttmp;
-					}
 
-					return $objecttmp;
-				}
-			} else {
-				return $objecttmp;	// returned an object without fetch
-			}
-		} else {
-			return -1;
+/**
+ * Fetch an object from its id and element_type
+ * Inclusion of classes is automatic
+ *
+ * @param	int     	$element_id 		Element id (Use this or element_ref but not both. If id and ref are empty, object with no fetch is returned)
+ * @param	string  	$element_type 		Element type ('module' or 'myobject@mymodule' or 'mymodule_myobject')
+ * @param	string     	$element_ref 		Element ref (Use this or element_id but not both. If id and ref are empty, object with no fetch is returned)
+ * @param	int<0,2>	$useCache 			If you want to store object in cache or get it from cache 0 => no use cache , 1 use cache, 2 force reload  cache
+ * @param	int			$maxCacheByType 	Number of object in cache for this element type
+ * @param	string		$errMsgCode 		Error message
+ * @return null|false|object 				object || null on not found || false if error
+ * @see getElementProperties()
+ * @see newObjectByElement()
+ */
+function loadObjectByElement($element_id, $element_type, $element_ref = '', $useCache = 0, $maxCacheByType = 10, &$errMsgCode = '')
```

Comment:
There is already a method to get an object from its id and type.
It is "fetchObjectByElement"